platform/linux-generic/include/odp_packet_continuous.h

```diff
@@ -0,0 +1,104 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright (c) 2025 Nokia
+ */
+
+#include <odp/api/packet.h>
+#include <odp_debug_internal.h>
+
+#ifndef ODP_PACKET_CONTINUOUS_H
+#define ODP_PACKET_CONTINUOUS_H
+
+/*
+ * These functions make it more convenient to read and write packet data
+ * when it is not known if there is a segment boundary in the packet in
+ * the area being read from or written to.
+ */
+
+/* State needed by the API */
+struct continuous_s;
+typedef struct continuous_s continuous_t;
+
+/* Maximum value of the len parameter that may be passed to the functions */
+#define CONTINUOUS_MAX_LEN 40
+
+/*
+ * Return a pointer to 'len' continuous bytes of packet data starting at
+ * L3 offset. If there is a segment boundary in the packet in the area of
+ * interest, packet data is copied to a temporary buffer in 'state' and the
+ * returned pointer points to that buffer.
+ *
+ * 'len' must not exceed CONTINUOUS_MAX_LEN.
+ *
+ * Returns a null pointer on error.
+ */
+static inline void *continuous_copy_l3(continuous_t *state, odp_packet_t pkt, uint32_t len);
+
+/*
+ * Copy back the copied and possibly modified packet data from 'state' to the
+ * original packet or do nothing if copying was not needed and packet data
+ * could be modified in-place. Does not need to be called if packet data was
+ * only read and not written to.
+ *
+ * Returns nonzero on error.
+ */
+static inline int continuous_copy_back(continuous_t *state);
```

Comment:
Internal headers typically seem to just define the functions directly if inline (instead of declare/define). Also, `_odp_` prefix to match the current style.

Even though name mangling is probably not an issue with these, internal headers seem to still wrap these in `extern "C"` block.